NPE in DEV console command

### Describe the bug

I started my app with `mvn clean quarkus:dev`, opened the DEV console with `:` then typed `postgres print-command` and got an NPE:

```
You are now in Quarkus Terminal. Your app is still running. Use `help` or tab completion to explore, `quit` or `q` to return to your application.
quarkus$ postgres print-command 
java.lang.NullPointerException
quarkus$ 
```

The normal terminal has more info:

```
2022-08-31 11:46:36,602 WARNING [org.aes.rea.Process] (Thread-77) Uncaught exception when executing the command: io.quarkus.devservices.postgresql.deployment.PsqlCommand@78eb1c73: java.lang.RuntimeException: java.lang.NullPointerException
	at org.aesh.command.impl.Executions$ExecutionImpl.execute(Executions.java:237)
	at org.aesh.readline.Process.run(Process.java:80)
Caused by: java.lang.NullPointerException
	at io.quarkus.devservices.postgresql.deployment.PsqlCommand.execute(PsqlCommand.java:28)
	at org.aesh.command.impl.operator.EndOperator.execute(EndOperator.java:41)
	at org.aesh.command.impl.Executions$ExecutionImpl.execute(Executions.java:213)
	... 1 more
```

[quarkus-dev-console-psql.zip](https://github.com/quarkusio/quarkus/files/9460437/quarkus-dev-console-psql.zip)


### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/devservices/postgresql/src/main/java/io/quarkus/devservices/postgresql/deployment/PsqlCommand.java
+++ b/extensions/devservices/postgresql/src/main/java/io/quarkus/devservices/postgresql/deployment/PsqlCommand.java
@@ -14,6 +14,8 @@ import io.quarkus.deployment.builditem.DevServicesLauncherConfigResultBuildItem;
 @CommandDefinition(name = "print-command", description = "Outputs the psql command to connect to the database")
 public class PsqlCommand implements Command {
 
+    private static final String JDBC_DATASOURCE__URL_KEY = "quarkus.datasource.jdbc.url";
+    private static final String REACTIVE_DATASOURCE__URL_KEY = "quarkus.datasource.reactive.url";
     private final DevServicesLauncherConfigResultBuildItem devServicesLauncherConfigResultBuildItem;
 
     public PsqlCommand(DevServicesLauncherConfigResultBuildItem devServicesLauncherConfigResultBuildItem) {
@@ -24,14 +26,25 @@ public class PsqlCommand implements Command {
     public CommandResult execute(CommandInvocation commandInvocation) throws CommandException, InterruptedException {
         //todo: should this work for non-dev services stuff as well? What about non-default datasources
         try {
-            URI url = new URI(
-                    devServicesLauncherConfigResultBuildItem.getConfig().get("quarkus.datasource.jdbc.url").substring(5));
-            int port = url.getPort();
-            String host = url.getHost();
+            URI uri;
+            if (devServicesLauncherConfigResultBuildItem.getConfig().containsKey(JDBC_DATASOURCE__URL_KEY)) {
+                uri = new URI(
+                        devServicesLauncherConfigResultBuildItem.getConfig().get(JDBC_DATASOURCE__URL_KEY)
+                                .substring("jdbc:".length()));
+            } else if (devServicesLauncherConfigResultBuildItem.getConfig().containsKey(REACTIVE_DATASOURCE__URL_KEY)) {
+                uri = new URI(
+                        devServicesLauncherConfigResultBuildItem.getConfig().get(REACTIVE_DATASOURCE__URL_KEY)
+                                .substring("vertx-reactive:".length()));
+            } else {
+                throw new RuntimeException("Unable to determine datasource URL");
+            }
+
+            int port = uri.getPort();
+            String host = uri.getHost();
             String pw = devServicesLauncherConfigResultBuildItem.getConfig().get("quarkus.datasource.password");
             commandInvocation.println("PGPASSWORD=" + pw + " psql --host=" + host + " --port=" + port + " --username="
                     + devServicesLauncherConfigResultBuildItem.getConfig().get("quarkus.datasource.username") + " "
-                    + url.getPath().substring(1));
+                    + uri.getPath().substring(1));
             return CommandResult.SUCCESS;
         } catch (URISyntaxException e) {
             throw new RuntimeException(e);

